packages/eslint-plugin/src/rules/no-require-imports.ts

```diff
@@ -1,27 +1,59 @@
 import type { TSESTree } from '@typescript-eslint/utils';
-import { ASTUtils } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES, ASTUtils } from '@typescript-eslint/utils';
 import { getScope } from '@typescript-eslint/utils/eslint-utils';
 
-import { createRule } from '../util';
+import * as util from '../util';
 
-export default createRule({
+type Options = [
+  {
+    allow: string[];
+  },
+];
+type MessageIds = 'noRequireImports';
+
+export default util.createRule<Options, MessageIds>({
   name: 'no-require-imports',
   meta: {
     type: 'problem',
     docs: {
       description: 'Disallow invocation of `require()`',
     },
-    schema: [],
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allow: {
+            type: 'array',
+            items: { type: 'string' },
+            description: 'Patterns of import paths to allow requiring from.',
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
     messages: {
       noRequireImports: 'A `require()` style import is forbidden.',
     },
   },
-  defaultOptions: [],
-  create(context) {
+  defaultOptions: [{ allow: [] }],
+  create(context, options) {
+    const allowPatterns = options[0].allow.map(
+      pattern => new RegExp(pattern, 'u'),
+    );
+    function isImportPathAllowed(importPath: string): boolean {
+      return allowPatterns.some(pattern => importPath.match(pattern));
+    }
     return {
       'CallExpression[callee.name="require"]'(
         node: TSESTree.CallExpression,
       ): void {
+        if (
+          node.arguments[0]?.type === AST_NODE_TYPES.Literal &&
+          typeof node.arguments[0].value === 'string' &&
+          isImportPathAllowed(node.arguments[0].value)
+        ) {
```

Comment:
does it work with string constants?

```TS
const path = 'package.json';
const pkg = require(path);
```

Reply:
No, that needs to make the rule type-aware, or otherwise implement a go-to-definition mechanism, which I'm not sure we have the precedent. I'm also not sure if it happens more than a handful of times.